S/SHTns/build_tarballs.jl

```diff
@@ -0,0 +1,42 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "SHTns"
+version = v"3.5.1"
+
+# Collection of sources required to complete build
+sources = [
+    ArchiveSource("https://bitbucket.org/nschaeff/shtns/downloads/shtns-$(version).tar.gz", "77f8a33b94df8786d2ce9b95cbfbe548f00443625b310b8c64012b22c8a7394f")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/shtns/
+export CFLAGS="-fPIC"
+./configure --prefix=${prefix} --build=${MACHTYPE} --host=${target}
+make -j${nproc}
+make install
+mkdir -p ${libdir}
+gcc -shared -o "${libdir}/libshtns.${dlext}" *.o -lfftw3
```

Comment:
```suggestion
cc -shared -o "${libdir}/libshtns.${dlext}" *.o -lfftw3
```

Reply:
There is a bunch of undefined symbols, e.g. `__sy11_l`. Similar symbols are found in `sht_kernels_s.so`, e.g. __sy12_l` .
The missing symbols are present in this file when I build on linux. I also checked that the generated code is the same, so this must be something related to CPP macros and conditional compilation. This is quite a bit beyond my expertise.

Reply:
That's why I'm suggesting to compile with `cc` (which on macOS is Clang, not GCC) instead of `gcc`, since the rest of the compilation was done with Clang, not GCC.  Mixing up compilers like that isn't good, and may well be the culprit here.